src/app/auth/profiles/profiles.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { AppConfig } from '../../../environments/environment';
+import { MatDialog } from '@angular/material';
+import {
+  JsonParseErrorDialogComponent
+} from './json-parse-error-dialog/json-parse-error-dialog.component';
+import ProfilesJson from './profiles.json';
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  username: string;
+  password: string;
+  encodedText: string;
+}
+
+/**
+ * This component is responsible allowing the user to select a Profile
+ * in the authentication page.
+ */
+
+@Component({
+  selector: 'app-profiles',
+  templateUrl: './profiles.component.html',
+  styleUrls: ['./profiles.component.css']
+})
+export class ProfilesComponent implements OnInit {
+
+  profiles: Profile[] = undefined; // List of profiles taken from profiles.json
+  blankProfile: Profile = {profileName: '', password: '', username: '',
+    encodedText: ''}; // A blank profile to reset values
+
+  private readonly fs = require('fs');
+
+  // path of profile.json (same folder as app for production)
+  private filePath: string = __dirname.split('CATcher', 1)[0]
+    .concat('profiles.json');
```

Comment:
can we complete the `.concat` part in the same line. I don't see the need to continue to a new line.